### PR TITLE
Copter: make landing_gear_should_be_deployed a base-class method

### PR DESCRIPTION
**Pull request by peterbarker**

### CODE REVIEW

File: ArduCopter/mode_auto.cpp [hunk 1]
```diff
@@ -411,7 +411,7 @@ void Copter::ModeAuto::land_run()
     _copter.land_run_vertical_control();
 }
 
-bool Copter::ModeAuto::landing_gear_should_be_deployed()
+bool Copter::ModeAuto::landing_gear_should_be_deployed() const
 {
     switch(_mode) {
     case Auto_Land:
```
Comment (OXINARF) [inline ArduCopter/mode_auto.cpp]: GitHub doesn't allow me to comment below. Wouldn't it be better in Auto_RTL to do `return _copter.mode_rtl.landing_gear_should_be_deployed()`?

Comment (peterbarker) [reply ArduCopter/mode_auto.cpp]: Not really related to this PR.

But still a great cleanup.  Added a patch on top.

**Comment by peterbarker** [comment]
On Wed, 27 Dec 2017, Francisco Ferreira wrote:

> @peterbarker Can you just fix the second commit sub-message? It's missing
> two characters: he -> the; alo -> also. I'll do it tomorrow if you don't do
> it sooner.

Done.

**Comment by OXINARF** [comment]
Merged, thanks!

**Pull request merged**